packages/rest/src/utils/agent.ts

```diff
@@ -1,80 +1,163 @@
+import type { ModulesMap } from '@aries-framework/core'
+import type { IndyVdrPoolConfig } from '@aries-framework/indy-vdr'
+
+import {
+  AnonCredsCredentialFormatService,
+  AnonCredsProofFormatService,
+  V1CredentialProtocol,
+  AnonCredsModule,
+  LegacyIndyCredentialFormatService,
+  LegacyIndyProofFormatService,
+  V1ProofProtocol,
+} from '@aries-framework/anoncreds'
+import { AnonCredsRsModule } from '@aries-framework/anoncreds-rs'
+import { AskarModule } from '@aries-framework/askar'
 import {
+  V2CredentialProtocol,
+  V2ProofProtocol,
+  Agent,
   AutoAcceptCredential,
   AutoAcceptProof,
-  Agent,
-  ConnectionInvitationMessage,
+  ConnectionsModule,
+  CredentialsModule,
   HttpOutboundTransport,
   LogLevel,
-  utils,
+  MediatorModule,
+  ProofsModule,
 } from '@aries-framework/core'
+import { IndyVdrAnonCredsRegistry, IndyVdrModule } from '@aries-framework/indy-vdr'
 import { agentDependencies, HttpInboundTransport } from '@aries-framework/node'
+import { anoncreds } from '@hyperledger/anoncreds-nodejs'
+import { ariesAskar } from '@hyperledger/aries-askar-nodejs'
+import { indyVdr } from '@hyperledger/indy-vdr-nodejs'
 import path from 'path'
 
 import { TsLogger } from './logger'
 import { BCOVRIN_TEST_GENESIS } from './util'
 
+export interface RestAgentModules extends ModulesMap {
+  connections: ConnectionsModule
+  proofs: ProofsModule<[V1ProofProtocol, V2ProofProtocol<[LegacyIndyProofFormatService, AnonCredsProofFormatService]>]>
+  credentials: CredentialsModule<[V1CredentialProtocol, V2CredentialProtocol]>
+  anoncreds: AnonCredsModule
+}
+
+export type RestAgent<
+  modules extends RestAgentModules = {
+    connections: ConnectionsModule
+    proofs: ProofsModule<
+      [V1ProofProtocol, V2ProofProtocol<[LegacyIndyProofFormatService, AnonCredsProofFormatService]>]
+    >
+    credentials: CredentialsModule<[V1CredentialProtocol, V2CredentialProtocol]>
+    anoncreds: AnonCredsModule
+  }
+> = Agent<modules>
+
```

Comment:
Can we not infer these types from the modules? If not, this is perfectly fine but if we can do something like `type RestAgent = Agent<ReturnType<getAgentModules>>` (I think) that would be better.